dts/arm/st/wb0/stm32wb09.dtsi

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/wb0/stm32wb05.dtsi>
+
+/**
+ * STM32WB09 is mostly identical to STM32WB05.
+ * The only differences are RAM size, flash size and TRNG IP.
+ */
+/ {
+	soc {
+		compatible = "st,stm32wb09", "st,stm32wb0", "simple-bus";
+
+		flash: flash-controller@40001000 {
+			flash0: flash@10040000 {
+				reg = <0x10040000 DT_SIZE_K(512)>;
+			};
+		};
+	};
+
+	sram0: memory@20000000 {
+		reg = <0x20000000 DT_SIZE_K(64)>;
+	};
+};
```

Comment:
Nitpick:
Move memory defnitions in `dts/arm/st/wb0/stm32wb09Xe.dtsi` file.
No consequence and eases automated file parsing if need arises

Apply uniformously

Reply:
> No consequence

It does lead to duplication as both `stm32wb06Xc` and `stm32wb07Xc` will now define flash size. Other than that, I agree.

Change will be done.